Implemented back-end game logic

PThompson284 opened the pull request:
Implements the rules for the game jungle in java
closes #19 #20

PThompson284 committed:
Basic structure of game and movement logic

PThompson284 committed:
Added ability to move pieces and start of movement validity logic.

PThompson284 committed:
Stylization changes

PThompson284 committed:
Stylization changes

PThompson284 committed:
Fix issue with turns not changing

PThompson284 committed:
Merge branch 'pt_dev' of https://github.com/DavidWellsTheDeveloper/cs414-f18-001-Team.name into pt_dev

PThompson284 committed:
Updated game to receive commands one tile at a time instead of having both tiles of the move known at once.

PThompson284 committed:
Update game logic for movement to handle water, friendly, and enemy units.

PThompson284 committed:
updated game logic and added comments

brian-crane committed:
Adding new JUnit Test for bcrane and Java-fying Game variables

PThompson284 committed:
Added game logic for lion/tiger leaping
Added game logic for win conditions

PThompson284 committed:
Merge branch 'pt_dev' of https://github.com/DavidWellsTheDeveloper/cs414-f18-001-Team.name into pt_dev

DavidWellsTheDeveloper reviewed (commented):
Great start! A lot of this is good. Before getting too deep into the game logic though, I think we should break this down into smaller isolated classes. Let's use some design principles we've been working on in lecture as far as object planning, and system design. I'm happy to do this with anyone who is interested. If you want to merge this into development I would approve that.

PThompson284 committed:
Expand on testing for game logic

PThompson284 committed:
Fix handling of traps (friendly vs enemy)
Expanded on tests

PThompson284 committed:
Expand on test output

PThompson284 committed:
Change how commands are received to be (row, col) instead of (col, row) to be consistent with internal board structure

brian-crane committed:
Changing GameTest to use normal JUnit Annotations and removing GameTest2

brian-crane committed:
Changing GameTest to use normal JUnit Annotations and removing GameTest2

brian-crane committed:
Adding GameUtils class to help visualize board
Adding new JUnit test to make sure you cant move pieces onto existing pieces
Adding print statement to MakeMove method in Game class

brian-crane committed:
Resolving merge conflict with pt_dev by updating sendInput method sig

brian-crane committed:
Merge remote-tracking branch 'remotes/origin/pt_dev' into bc_pt_dev

# Conflicts:
#	server/src/Game.java

brian-crane committed:
Updating Game.java to show move sequence, and to use GameUtils

Also added changes to JUnit test testBlueLionNotAbleToCrossItsOwnPieces to use new sendMove method signature

brian-crane committed:
Merge pull request #40 from DavidWellsTheDeveloper/bc_pt_dev

Bc pt dev
